executor/ddl.go

```diff
@@ -76,13 +76,47 @@ func (e *DDLExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	}
 	e.done = true
 
+	isLocalTempTable := func(keyWord ast.TemporaryKeyword) bool {
+		return keyWord == ast.TemporaryLocal
+	}
+	dropLocalTempTable := func(originTbNames []*ast.TableName) []*ast.TableName {
+		sessVars := e.ctx.GetSessionVars()
+		sessVarsTempTable := sessVars.LocalTemporaryTables
+		if sessVarsTempTable == nil {
+			return originTbNames
+		}
+		filteredTbNames := make([]*ast.TableName, 0)
+		localTempTables := sessVarsTempTable.(*infoschema.LocalTemporaryTables)
+		for tbIdx := range originTbNames {
+			if localTempTables.RemoveTable(originTbNames[tbIdx].Schema, originTbNames[tbIdx].Name) {
+				continue
+			}
+			filteredTbNames = append(filteredTbNames, originTbNames[tbIdx])
+		}
+		return filteredTbNames
+	}
+
 	// For each DDL, we should commit the previous transaction and create a new transaction.
-	// An exception is create local temporary table.
-	if s, ok := e.stmt.(*ast.CreateTableStmt); ok {
-		if s.TemporaryKeyword == ast.TemporaryLocal {
+	// Following cases are exceptions
+	switch s := e.stmt.(type) {
+	case *ast.CreateTableStmt:
+		if isLocalTempTable(s.TemporaryKeyword) {
 			return e.createSessionTemporaryTable(s)
 		}
+	case *ast.DropTableStmt:
+		if !s.IsView {
+			s.Tables = dropLocalTempTable(s.Tables)
```

Comment:
Assume we have two tables, `tmp1` as a local temporary table and `normal1` as a normal table. What happens if tmp1 exists and normal1 not exists? I guess it will return error but tmp1 still dropped.

Maybe its better to drop temp tables after dropping normal table when normal table exists.